tools/mkbuildoptglobals.py

```diff
@@ -609,8 +616,19 @@ def main():
     global build_opt_signature
     global docs_url
     global debug_enabled
+    global default_locale
     num_include_lines = 1
 
+    # Given that GCC will handle lines from an @file as if they were on
+    # the command line. I assume that the contents of @file need to be encoded
+    # to match that of the shell running GCC runs. I am not 100% sure this API
+    # gives me that, but it appears to work.
+    #
+    # However, elsewhere when dealing with source code we continue to use 'utf-8',
+    # ref. https://gcc.gnu.org/onlinedocs/gcc-4.1.2/cpp/Character-sets.html
```

Comment:
should link to latest?
https://gcc.gnu.org/onlinedocs/cpp/Character-sets.html

pending locale explanation from comment. afaik, we just have to use what system uses for fs, why I mentioned sys.getfilesystemencoding(); mingw -> gcc layer is expectedly using fopen() as-is, so there have to be some kind of raw bytes in the correct encoding for sys api to work

Reply:
```
locale.getdefaultlocale()                ('en_US', 'cp1252')
sys.getfilesystemencoding()              utf-8
sys.getdefaultencoding()                 utf-8
locale.getpreferredencoding(False)       cp1252
sys.stdout.encoding                      cp1252
locale.setlocale(locale.LC_ALL, None)    C
locale.getlocale()                       (None, None)
locale.setlocale(locale.LC_ALL, '')      English_United States.1252
locale.getlocale()                       ('English_United States', '1252')
```

> \<stdin\>:1: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead

You must be using a different version of python than the release.
For   `locale.getencoding()`
I get: `AttributeError: module 'locale' has no attribute 'getencoding` 

It is a bit frustrating. Any solution I find on the Internet either doesn't work or will shortly be deprecated and stop working.
I guess `setlocale(), getencoding() and getlocale() ` is supposed to be the right way. I couldn't find any examples using that set. And `getencoding()` doesn't appear to exist in 3.10.6. The docs say they added it in 3.11. They don't leave much time before deprecating stuff.

Reply:
Ugh. So, looking at patch that adds it
https://github.com/python/cpython/commit/677320348728ce058fa3579017e985af74a236d4
Surprising side-effect from `-X utf-8` which we use for sizes.py, that (sort of) explains the difference with the encoding='locale' for TextIO <-> subprocess
(also casually mentioned in docs)

Suppose this difference could just we wrapped with sys.version check at a global level to avoid possible future warnings. It is the right way, just heavily modified version to version :/
```python
>>> sys.version_info
sys.version_info(major=3, minor=7, micro=2, releaselevel='final', serial=0)
>>> sys.version_info >= (3, 7)
True
>>> sys.version_info >= (3, 11)
False
```